datalad/core/local/run.py

```diff
@@ -573,10 +585,15 @@ def run_command(cmd, dataset=None, inputs=None, outputs=None, expand=None,
     # relative paths, and (3) happen within a chpwd(pwd) context.
     if not inject:
         with chpwd(pwd):
-            for res in prepare_inputs(ds_path, inputs, extra_inputs):
+            for res in prepare_inputs(
+                    ds_path,
+                    [] if assume_ready in ["inputs", "both"] else inputs,
```

Comment:
so we will still take care about "extra_inputs"... is that intended?

Reply:
> so we will still take care about "extra_inputs"... is that intended?

Yes, from the line below:

https://github.com/datalad/datalad/blob/08f0a004723b1d91c502dac36812cedec56c2a8d/datalad/core/local/run.py#L590-L593